Allow class inheritance determination for enums (#16)
Crell/AttributeUtils: README.md

In this example, `$displayInfoA->type` will be an instance of `Screen`, `$displayInfoB->type` will be an instance of `Audio`, and `$displayInfoC->type` will be `null`.

### Interface using attributes

Interfaces may also implement attributes. This is useful for cases in which you want to define a type map at the interface level.

> Note: in this case, make sure the class implements the `Inheritable` interface, to return the expected values.

Reviewer: ```suggestion
Note: in this case, make sure the attribute class implements the `Inheritable` interface, so that implementing classes can inherit it.
```

In this example, `$displayInfoA->type` will be an instance of `Screen`, `$displayInfoB->type` will be an instance of `Audio`, and `$displayInfoC->type` will be `null`.

### Interface using attributes

Interfaces may also implement attributes. This is useful for cases in which you want to define a type map at the interface level.

Note: in this case, make sure the class implements the `Inheritable` interface, to return the expected values.

```php
use Crell\AttributeUtils\Inheritable;

#[Attribute(Attribute::TARGET_CLASS)]

> final readonly class TypeMap implements Inheritable

Reviewer: Serde TypeMaps were the reason I added this feature, but I don't think they make a particularly good example.  It requires a lot of context to understand why you'd want to do it.  For the README, we should have a much simpler and more understandable example.  Remember, not everyone using AU is using Serde.  (I hope... 😄 )

Author: Indeed, that is a reasonable take. I used the Serde TypeMaps since that is what I'm using as reference for these changes.

The TypeMap also helped me learn that if an attribute does not implement `Inheritable` the result is different than expected, which is something I believe the Readme needs to account for.

With that said, I will try to write an example that is not tied to the TypeMaps.

Author: Hopefully the new example I pushed makes more sense and requires less context.

In this example, `$displayInfoA->type` will be an instance of `Screen`, `$displayInfoB->type` will be an instance of `Audio`, and `$displayInfoC->type` will be `null`.

### Interface using attributes

> Interfaces may also implement attributes. This is useful for cases in which you want to define a type map at the interface level:

Reviewer: ```suggestion
Interfaces may also implement attributes. This is useful for cases in which you want to define a type map at the interface level.
```

There's another line between this and the example, so it shouldn't be a colon.

Note: in this case, make sure the class implements the `Inheritable` interface, to return the expected values.

```php

> 

Reviewer: ```suggestion
```
